pkg/controller/nodegroup/nodegroup_controller.go

```diff
@@ -341,31 +339,39 @@ func (r *ReconcileNodeGroup) updateNodeGroupStack(cfnSvc cloudformationiface.Clo
 	return err
 }
 
-func parseCFNParameterFromCRD(ngSpec clusterv1alpha1.NodeGroupSpec) []*cloudformation.Parameter {
-	if ngSpec.Instance == nil {
-		return nil
-	}
+func parseCFNParameterFromCRD(ng *clusterv1alpha1.NodeGroup) []*cloudformation.Parameter {
 
 	var parameter []*cloudformation.Parameter
 
-	if ngSpec.Instance.InstanceType != nil {
-		parameter = append(parameter, &cloudformation.Parameter{
-			ParameterKey:   aws.String("NodeInstanceType"),
-			ParameterValue: ngSpec.Instance.InstanceType,
-		})
-	}
+	if ng.Spec.Instance != nil {
+		if ng.Spec.Instance != nil {
+			if ng.Spec.Instance.InstanceType != nil {
+				parameter = append(parameter, &cloudformation.Parameter{
+					ParameterKey:   aws.String("NodeInstanceType"),
+					ParameterValue: ng.Spec.Instance.InstanceType,
+				})
+			}
 
-	if ngSpec.Instance.MaxInstanceCount != nil {
-		parameter = append(parameter, &cloudformation.Parameter{
-			ParameterKey:   aws.String("NodeAutoScalingGroupMaxSize"),
-			ParameterValue: aws.String(strconv.Itoa(*ngSpec.Instance.MaxInstanceCount)),
-		})
+			if ng.Spec.Instance.MaxInstanceCount != nil {
+				parameter = append(parameter, &cloudformation.Parameter{
+					ParameterKey:   aws.String("NodeAutoScalingGroupMaxSize"),
+					ParameterValue: aws.String(strconv.Itoa(*ng.Spec.Instance.MaxInstanceCount)),
+				})
+			}
+
+			if ng.Spec.Instance.EBSVolumeSize != nil {
+				parameter = append(parameter, &cloudformation.Parameter{
+					ParameterKey:   aws.String("NodeVolumeSize"),
+					ParameterValue: aws.String(strconv.Itoa(*ng.Spec.Instance.EBSVolumeSize)),
+				})
+			}
+		}
 	}
 
-	if ngSpec.Instance.EBSVolumeSize != nil {
```

Comment:
Are we removing ability to update based on volume size change? or it is just diff is messing up?

Reply:
This looks good in my branch, diff is messing up, its removing and adding it few lines above